ui/src/map/map-store.ts

```diff
@@ -25,6 +25,20 @@ export function getEmptyShownItems(): ShownItems {
     };
 }
 
+export const isLayerEnabledByProxy = (
+    menuItemName: MapLayerMenuItemName,
+    visibleLayers: MapLayerName[],
+): boolean => {
+    const layersFromMenuItem = layerMenuItemMapLayers[menuItemName];
+    const key = Object.keys(relatedMapLayers).find(
+        (key) =>
+            relatedMapLayers[key as MapLayerName]?.find((layer) =>
+                layersFromMenuItem.includes(layer),
+            ),
+    );
+    return visibleLayers.some((layer) => layer === key);
+};
+
```

Comment:
🥩 2

Tästä koodista tuli vähän nurinkurista sillä pitää kävellä aiemmin oletettavasti yksisuuntaisiksi oletettuja mäppäyksiä taaksepäin (siksi mm. tuo `Object.keys`). Tilanteena tämä on kuitenkin sen verran harvinainen, että en katsonut järkeväksi lähteä reworkkaamaan tuota `relatedMapLayers`-rakennetta johonkin helpommin kaksisuuntaisesti kuljettavaan muotoon.